akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -203,8 +203,8 @@ object HttpEntity {
    *
    * If the given `chunkSize` is -1 the default chunk size is used.
    */
-  def apply(contentType: ContentType, file: File, chunkSize: Int = -1): UniversalEntity = {
-    val fileLength = file.length
+  def apply(contentType: ContentType, file: Path, chunkSize: Int = -1): UniversalEntity = {
```

Comment:
I can't unfortunately do this:

```Scala
def apply(contentType: ContentType, file: Path, chunkSize: Int = -1): UniversalEntity = ???
def apply(contentType: ContentType, file: File, chunkSize: Int = -1): UniversalEntity = ???

// in object HttpEntity, multiple overloaded alternatives of method apply define default arguments.
```

What I can do instead is:

```Scala
def apply(contentType: ContentType, file: Path, chunkSize: Int): UniversalEntity = ???
def apply(contentType: ContentType, file: Path): UniversalEntity = ???
def apply(contentType: ContentType, file: File, chunkSize: Int = -1): UniversalEntity = ???
```

Or introduce a new method like `fromPath`. What you would prefer?